### PR TITLE
Add view location lookup method to AssetModel.

### PR DESCRIPTION
**Pull request by kaecyra**
Also adds search() to Gdn_Factory and viewHandlers() and viewExtensions() methods to AssetModel.

### CODE REVIEW

File: applications/dashboard/models/class.assetmodel.php [hunk 1]
```diff
@@ -95,154 +95,153 @@ public function serveCss($ThemeType, $Filename) {
         require_once PATH_LIBRARY."/vendors/Minify/lib/Minify/CSS.php";
 
         ob_start();
-        echo "/* CSS generated for etag: $CurrentETag.\n *\n";
+        echo "/* CSS generated for etag: $currentETag.\n *\n";
 
-        $NotFound = array();
-        $Paths = $this->GetCssFiles($ThemeType, $Basename, $ETag, $NotFound);
+        $notFound = [];
```
Comment (mtschirs) [inline applications/dashboard/models/class.assetmodel.php]: AFAIK [] is PHP >= 5.4 only

File: applications/dashboard/models/class.assetmodel.php [hunk 2]
```diff
@@ -95,154 +95,153 @@ public function serveCss($ThemeType, $Filename) {
...
-            $Css = val('Css', $Options);
-            if (!$Css) {
-                $Css = file_get_contents($Path);
+            $css = val('Css', $options);
+            if (!$css) {
+                $css = file_get_contents($path);
             }
 
-            $Css = Minify_CSS::minify($Css, array(
+            $css = Minify_CSS::minify($css, [
```
Comment (mtschirs) [inline applications/dashboard/models/class.assetmodel.php]: same issue, [] is PHP >= 5.4 only

File: applications/dashboard/models/class.assetmodel.php [hunk 3]
```diff
@@ -95,154 +95,153 @@ public function serveCss($ThemeType, $Filename) {
...
+     * @param $basename
+     * @param $eTag
+     * @param null $notFound
      * @return array
      * @throws Exception
      */
-    public function getCssFiles($ThemeType, $Basename, $ETag, &$NotFound = null) {
-        $NotFound = array();
+    public function getCssFiles($themeType, $basename, $eTag, &$notFound = null) {
+        $notFound = [];
```
Comment (mtschirs) [inline applications/dashboard/models/class.assetmodel.php]: same issue, [] is PHP >= 5.4 only

**Comment by kaecyra** [comment]
@mtschirs 
https://github.com/vanilla/vanilla/blob/master/README.md

> Vanilla 2.2 requires PHP 5.3. Running master branch requires PHP 5.4, as will future official releases.

**Comment by mtschirs** [comment]
> Running master branch requires PHP 5.4

Nice, I thought 2.2 was still PHP >= 5.3 :+1:

**Comment by mtschirs** [comment]
Just for clarification:
> Vanilla 2.2 requires PHP 5.3. Running `master` branch requires PHP 5.4,

Just for clarification:
> Vanilla 2.2 requires PHP 5.3. Running `master` branch requires PHP 5.4,

Does this mean that contributions to the master branch such as this one will not become part of the official 2.2 relase?

**Comment by linc** [comment]
That's correct. I will be having some quality time with the 2.2 branch to cherry pick security fixes before it goes gold. We'll also likely be moving to 2.3 much quicker than we have historically taken between releases.

**Pull request merged**